quiche/src/lib.rs

```diff
@@ -2030,7 +2030,17 @@ impl Connection {
             ) {
                 Ok(v) => v,
 
-                Err(Error::Done) => left,
+                Err(Error::Done) => {
+                    // If the packet can't be processed or decrypted, check if it's a stateless
+                    // reset.
+                    if self.is_stateless_reset(&buf[len - left..len]) {
+                        trace!("{} packet is a stateless reset", self.trace_id);
+                        let now = time::Instant::now();
+                        let path = self.paths.get_active()?;
+                        self.draining_timer = Some(now + (path.recovery.pto() * 3));
```

Comment:
I don't think we need to drain the connection on stateless reset (if the peer sends a reset it's unlikely it will send any more packets), so you can just set `self.closed = true` here instead.